### PR TITLE
chore: migrate constants to TS

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

This makes it more straightforward to compile the new api into JS. Without these changes it will try to emit constants.js and fail with
```
error TS5055: Cannot write file '.../freeCodeCamp/config/constants.js' because it would overwrite input file.
```

### CODE REVIEW

**Review by ShaunSHamilton — APPROVED** [review]
Does not break stuff 👍 _Except the linter, because the new .js has not been ignored_

**Comment by naomi-lgbt** [comment]
Since we're using TS now do we want to make these all read-only?

**Comment by ojeytonwilliams** [comment]
> Since we're using TS now do we want to make these all read-only?

I pulled on the thread; maybe a little too hard. They're readonly now, though :cold_sweat:

**Comment by naomi-lgbt** [comment]
```console
    Expected: 671
    Received: 643
```
    
Welp.

**Comment by ojeytonwilliams** [comment]
Oops :face_exhaling: Fixed.

**Review by ShaunSHamilton — APPROVED** [review]
Even better 👍

File: config/constants.ts [hunk 1]
```diff
@@ -1,10 +1,33 @@
...
+  'v',
+  'w',
+  'x',
+  'y',
+  'z'
+] as const;
 
-for (let i = 0; i < 26; i++) {
-  alphabet += String.fromCharCode(97 + i);
-}
```
Comment (raisedadead) [inline config/constants.ts:5]: What's going to prevent someone from undoing this in the future - people try to act and code smart all the time, you know?

Comment (ojeytonwilliams) [reply config/constants.ts:5]: Nothing at the moment. Do you want a test for the alphabet?

Comment (ojeytonwilliams) [reply config/constants.ts:5]: It's not no obvious how to directly test this, but the last commit should make it clear that the blocklistedUsernames array is supposed to only contain literals.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**